pkg/pushsync/pushsync.go

```diff
@@ -365,13 +395,23 @@ func (ps *PushSync) pushToClosest(ctx context.Context, ch swarm.Chunk, retryAllo
 			if r.err != nil && r.attempted {
 				ps.metrics.TotalFailedSendAttempts.Inc()
 
-				// if the node has warmed up AND no other closer peer has been tried
-				if time.Now().After(ps.warmupPeriod) && !ps.skipList.HasChunk(ch.Address()) && ps.topologyDriver.IsWithinDepth(ch.Address()) {
-					ps.skipList.Add(peer, ch.Address(), skipPeerExpiration)
+				// if we have received 1 more results than peers attempted, return
+				// this means all attempts including establishing storage have already failed
+				if results > attempts {
+					return nil, ErrNoPush
 				}
 			}
 		case <-ctx.Done():
 			return nil, ctx.Err()
+		case <-sendReceipt:
+			// to make the loop wait for the one extra result because of this one time interruption, increase attempt exhaustion limits
+			allowedRetries++
+			i++
```

Comment:
maybe a `goto` statement to  a label immediately after the `for` loop could also work?

Reply:
could work, intention was to avoid goto statement so loop flow is easier to reason about

Reply:
or `goto` to top of the `select` which gets called after `go ps.establishStorage(ch, origin, resultC)`